### PR TITLE
db-sources:sort primary key based on key-seq

### PR DESCRIPTION
**Pull request by subodh1810**
Issue : https://github.com/airbytehq/airbyte/issues/28699

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  source-mysql test report (commit [`ed48150187`](https://github.com/airbytehq/airbyte/pull/29346/commits/ed48150187c7f2287fdf7faf2c6f22a168617c4a)) - ❌

⏲️  Total pipeline duration: 21mn09s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mysql/metadata.yaml     | ✅        |
| Connector version semver check                                          | ✅        |
| QA checks                                                               | ✅        |
| Build connector tar                                                     | ✅        |
| Build source-mysql docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mysql:integrationTest | ❌        |
| Acceptance tests                                                        | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sort-pkey-key-seq-basis/1691700305/ed48150187c7f2287fdf7faf2c6f22a168617c4a/source-mysql/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ed48150187c7f2287fdf7faf2c6f22a168617c4a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mysql test
```

File: airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java [hunk 1]
```diff
@@ -146,13 +148,14 @@ protected List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(fina
    *
    * @return a map by StreamName to associated list of primary keys
    */
-  private static Map<String, List<String>> aggregatePrimateKeys(final List<SimpleImmutableEntry<String, String>> entries) {
+  @VisibleForTesting
+  public static Map<String, List<String>> aggregatePrimateKeys(final List<PrimaryKeyAttributesFromDb> entries) {
```
[the remark this answers is not in the document]
Comment (akashkulk) [reply airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java:152]: we should, but I double there will be too much of a side effect here (just changing the order of keys)

[the remark this answers is not in the document]
Comment (subodh1810) [reply airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java:152]: Already ran! I dont expect anything to change.

**Review by akashkulk — APPROVED** [review]
Approving - we should just run tests for relevant connectors (mssql/mysql/postgres) just to ensure they don't check for order of keys in the catalog or something

**Review by prateekmukhedkar — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-mysql test report (commit [`1abf96e853`](https://github.com/airbytehq/airbyte/pull/29346/commits/1abf96e853453d607a702e1e5568cdec88027c92)) - ❌

⏲️  Total pipeline duration: 20mn32s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mysql/metadata.yaml     | ✅        |
| Connector version semver check                                          | ✅        |
| QA checks                                                               | ✅        |
| Build connector tar                                                     | ✅        |
| Build source-mysql docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mysql:integrationTest | ❌        |
| Acceptance tests                                                        | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/sort-pkey-key-seq-basis/1692124429/1abf96e853453d607a702e1e5568cdec88027c92/source-mysql/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1abf96e853453d607a702e1e5568cdec88027c92)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mysql test
```

**Comment by octavia-approvington** [comment]
Enough! Let's do this <br/> goooooo!

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**